pallets/gear/src/manager.rs

```diff
@@ -232,36 +232,44 @@ where
                     refund,
                 );
             } else {
-                log::error!(
-                    "Associated gas tree for message aren't able to be consumed: {:?}",
+                log::warn!(
+                    target: "runtime::gear",
+                    "Message {:?} consumed, gas hasn't yet been returned to the external origin",
                     message_id
                 );
             }
         } else {
             log::error!(
-                "Message does not have associated gas tree: {:?}",
+                target: "runtime::gear",
+                "Message doesn't have an associated gas tree: {:?}",
                 message_id
             );
         }
     }
     fn send_message(&mut self, message_id: MessageId, message: Message) {
         let message_id = message_id.into_origin();
-        let message: common::Message = message.into();
-
-        log::debug!("Message sent (from: {:?}): {:?}", message_id, message);
+        let mut message: common::Message = message.into();
 
-        if let Some(mut gas_tree) = ValueView::get(GAS_VALUE_PREFIX, message_id) {
-            let _ = gas_tree.split_off(message.id, message.gas_limit);
-        } else {
-            log::error!(
-                "Message does not have associated gas tree: {:?}",
-                message_id
-            );
-        }
+        log::debug!("Sending message {:?} from {:?}", message, message_id);
 
         if common::program_exists(message.dest) {
-            common::queue_message(message);
+            if let Some(mut gas_tree) = ValueView::get(GAS_VALUE_PREFIX, message_id) {
+                let _ = gas_tree.split_off(message.id, message.gas_limit);
+                common::queue_message(message);
+            } else {
+                log::error!(
```

Comment:
I don't think we should silently avoid queuing messages, even with `log::error`

either propagate error somewhere or still queue it

Reply:
We may at least add runtime event that message was going to be queued but couldn't